humility-core/src/hubris.rs

```diff
@@ -3648,22 +3872,32 @@ impl HubrisArchive {
                 registers: frameregs.clone(),
             });
 
-            let lr = *frameregs.get(&ARMRegister::LR).unwrap();
+            let ret_reg = *frameregs
+                .get(match self.arch {
+                    Some(goblin::elf::header::EM_ARM) => {
+                        &Register::Arm(ARMRegister::LR)
+                    }
+                    Some(goblin::elf::header::EM_RISCV) => {
+                        &Register::RiscV(RVRegister::RA)
+                    }
+                    _ => unimplemented!(),
+                })
+                .unwrap();
 
             //
             // If this is a kernel stack and we have hit an EXC_RETURN, we're
             // done.
             //
-            if task == HubrisTask::Kernel && (lr >> 28 == 0xf) {
+            if task == HubrisTask::Kernel && (ret_reg >> 28 == 0xf) {
```

Comment:
What is the `>> 28 == 0xf` check doing?

Reply:
On arm, when an exception occurs, the LR register gets filled with a magic value, this should detect it.  On riscv we will need a different way to do this, but since this function isn't actually used yet, I just added a TODO.

Reply:
Ah I see. SG.